lib/ansible/modules/packaging/os/zypper_repository.py

```diff
@@ -330,6 +333,10 @@ def main():
         'name': module.params['description'],
         'priority': module.params['priority'],
     }
+    # Get name from .repo file
+    if repodata['url'].endswith('.repo'):
+        rc, stdout, stderr = module.run_command('curl -s '+repodata['url'], check_rc=False)
```

Comment:
maybe just analyze what failing `zypper` command returns instead of downloading this twice?